overlord/snapstate/snapstate.go

```diff
@@ -579,11 +579,21 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	prev = setupSecurity
 
 	// finalize (wrappers+current symlink)
+	//
+	// For essential snaps that require reboots, 'link-snap' is currently marked as the edge of that reboot
+	// sequence. This means that we currently expect 'link-snap' to request the reboot and be the last task
+	// to run before the reboot takes place (for that lane/change). This task is assigned the edge 'MaybeRebootEdge'
+	// to indicate this.
```

Comment:
do we have tests that checks for this?

Reply:
AFAICT we don't have any tests that explicitly test that 'link-snap' and 'auto-connect' are marked as such

Reply:
yes, please, I think if we move the markers it needs to be a conscious decision, some corresponding tests with maybe in test comment could help